src/main/java/io/jenkins/plugins/traceable/ast/TraceableASTInitAndRunStepBuilder.java

```diff
@@ -457,92 +465,253 @@ private void runAndInitScan(TaskListener listener, Run<?, ?> run) {
             allEndPoint,
             replay
         };
-        runScript(scriptPath, args, listener, "runAndInitScan");
+
+        runScript(workspace, listener, scriptPath, args, "initAndRunScan");
     }
 
     // Stop the scan with the given scan ID.
-    private void abortScan(TaskListener listener) {
+    private void abortScan(FilePath workspace, TaskListener listener) {
         String scriptPath = "shell_scripts/stop_ast_scan.sh";
         String[] args = new String[] {traceableCliBinaryLocation, scanId};
-        runScript(scriptPath, args, listener, "abortScan");
+        runScript(workspace, listener, scriptPath, args, "abortScan");
     }
 
-    private void runScript(String scriptPath, String[] args, TaskListener listener, String caller) {
+    private void runScript(FilePath workspace, TaskListener listener, String scriptPath, String[] args, String caller) {
         try {
-            // Read the bundled script as string
-            String bundledScript = CharStreams.toString(
-                    new InputStreamReader(getClass().getResourceAsStream(scriptPath), Charsets.UTF_8));
-            // Create a temp file with uuid appended to the name just to be safe
+            String tempFilePath = workspace.act(new CopyScript(scriptPath));
+
+            if (caller.equals("initAndRunScan")) {
+                TraceableASTInitAndRunStepBuilder.scanId =
+                        workspace.act(new RunScript(listener, tempFilePath, args, caller));
+            } else if (caller.equals("abortScan")) {
+                workspace.act(new RunScript(listener, tempFilePath, args, caller));
+            }
+
+            deleteScript(workspace, listener, tempFilePath);
+
+        } catch (Exception e) {
+            log.error("Exception in running {} script : {}", scriptPath, e);
+            e.printStackTrace();
+        }
+    }
+
+    private void deleteScript(FilePath workspace, TaskListener listener, String scriptPath)
```

Comment:
The difference is only the ease of implementation. Instead of calling a `FileCallable` and then creating a `ProcessBuilder`, we can directly execute a command on the remote agent using `Launcher`. This makes it more easier to track the output and error stream.